types/react-autocomplete/index.d.ts

```diff
@@ -144,4 +144,16 @@ declare namespace Autocomplete {
         debug?: boolean;
     }
 }
-declare class Autocomplete extends Component<Autocomplete.Props> {}
+declare class Autocomplete extends Component<Autocomplete.Props> {
+    /**
+     * Autocomplete exposes a subset of `HTMLInputElement` properties to the parent component.
+     * They can be accessed through Autocomplete's `ref` prop.
+     */
+    blur: HTMLInputElement['blur'];
+    checkValidity: HTMLInputElement['checkValidity'];
+    click: HTMLInputElement['click'];
+    focus: HTMLInputElement['focus'];
+    select: HTMLInputElement['select'];
+    setCustomValidity: HTMLInputElement['setCustomValidity'];
+    setSelectionRange: HTMLInputElement['setSelectionRange'];
+}
```

Comment:
Small thing, but could you do `extends Component<Autocomplete.Props>, Pick<HTMLInputElement, 'blur' | 'checkValidity' | …> {}`?

Reply:
TypeScript classes can only extend a single class. Couldn't really find a clean workaround for that limitation.

Reply:
Ah, perhaps `implements Pick`?

Reply:
This mixin pattern can be accomplished using an [intersection type](http://www.typescriptlang.org/docs/handbook/advanced-types.html)

type AutoComplete = Component<Autocomplete.Props> & Pick<HTMLInputElement, 'blur' | 'checkValidity' | ...>;

Reply:
@OliverJAsh That's what TypeScript docs suggest in the Mixins section: https://www.typescriptlang.org/docs/handbook/mixins.html
But that felt too complicated (plus we can't declare functions in `.d.ts` files I believe)

@minestarks but we need `AutoComplete` to be declared as a `class` so that it can be used as a React Component. Or am I missing something here?

Reply:
Oops you're right, that was a hasty suggestion. Can't think of a good alternative at the moment.